modules/cnf-about-nic-support-irq-affinity-setting.adoc

```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// scalability_and_performance/cnf-low-latency-tuning.adoc
+
+:_content-type: CONCEPT
+[id="about_nic_support_irq_affinity_setting_{context}"]
+= About NIC support of IRQ affinity setting
+
+Some IRQ controllers lack support for IRQ affinity setting and will always expose all online CPUs as the IRQ mask.
+
+The following list by no means exhaustive are examples of drivers and hardware that Red Hat are aware lack support for IRQ re-balancing:
+
+* Some RAID controller drivers, such as `megaraid_sas`
+* Many non-volatile memory express (NVMe) drivers
+* Some LAN on motherboard (LOM) network controllers
+
+[NOTE]
+====
+LOMs are actually NICs controlled by NIC drivers.
+The reason they do not support IRQ affinity setting, is probably associated with factors such as the type of processor, the IRQ controller, or the circuitry connections in the motherboard.
+====
+
+These IRQ controllers effectively run on CPU 0. To find the effective affinity, log in to the host and run the following:
+
+[source,terminal]
+----
+$ find /proc/irq/ -name effective_affinity -exec sh -c 'i="$1"; mask=$(cat $i); file=$(echo $i); echo $file: $mask' _ {} \;
+----
+
+.Example output
+
+[source,terminal]
+----
+/proc/irq/0/effective_affinity: 1
+/proc/irq/1/effective_affinity: 8
+/proc/irq/2/effective_affinity: 0
+/proc/irq/3/effective_affinity: 1
+/proc/irq/4/effective_affinity: 2
+/proc/irq/5/effective_affinity: 1
+/proc/irq/6/effective_affinity: 1
+/proc/irq/7/effective_affinity: 1
+/proc/irq/8/effective_affinity: 1
+/proc/irq/9/effective_affinity: 2
+/proc/irq/10/effective_affinity: 1
+/proc/irq/11/effective_affinity: 1
+/proc/irq/12/effective_affinity: 4
+/proc/irq/13/effective_affinity: 1
+/proc/irq/14/effective_affinity: 1
+/proc/irq/15/effective_affinity: 1
+/proc/irq/24/effective_affinity: 2
+/proc/irq/25/effective_affinity: 4
+/proc/irq/26/effective_affinity: 2
+/proc/irq/27/effective_affinity: 1
+/proc/irq/28/effective_affinity: 8
+/proc/irq/29/effective_affinity: 4
+/proc/irq/30/effective_affinity: 4
+/proc/irq/31/effective_affinity: 8
+/proc/irq/32/effective_affinity: 8
+/proc/irq/33/effective_affinity: 1
+/proc/irq/34/effective_affinity: 2
+----
+
+Some drivers use `managed_irqs`, whose affinity is managed internally by the kernel and userspace cannot change the affinity. In some cases, these IRQs might be assigned to isolated CPUs.
```

Comment:
Is it possible to add here an explanation of how to know if an IRQ is managed_irq? From the article in access.redhat.com, either copying part of the content or linking to it.